cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -124,12 +124,12 @@ function reconcile_addons() {
   # Filter out `configured` message to not noisily log.
   # `created`, `pruned` and errors will be logged.
   log INFO "== Reconciling with deprecated label =="
-  ${KUBECTL} ${KUBECTL_OPTS} apply --namespace=${SYSTEM_NAMESPACE} -f ${ADDON_PATH} \
+  ${KUBECTL} ${KUBECTL_OPTS} apply -f ${ADDON_PATH} \
     -l ${CLUSTER_SERVICE_LABEL}=true,${ADDON_MANAGER_LABEL}!=EnsureExists \
     --prune=true --recursive | grep -v configured
 
   log INFO "== Reconciling with addon-manager label =="
-  ${KUBECTL} ${KUBECTL_OPTS} apply --namespace=${SYSTEM_NAMESPACE} -f ${ADDON_PATH} \
+  ${KUBECTL} ${KUBECTL_OPTS} apply -f ${ADDON_PATH} \
```

Comment:
I think this will work, but I need to confirm how `--prune` behave when multiple namespaces appear in manifest. Will give an update soon.

Reply:
Seems like it should work with the multiple namespaces case, kubectl records the visited namespaces for pruning:
https://github.com/kubernetes/kubernetes/blob/48388fec7eaad4ac8d84fbe20673ffacf41964a1/pkg/kubectl/cmd/apply.go#L391-L402

Also a bad news, at this moment the object types for pruning is hardcoded so don't think it will work well with the CRDs added by istio...
https://github.com/kubernetes/kubernetes/blob/48388fec7eaad4ac8d84fbe20673ffacf41964a1/pkg/kubectl/cmd/apply.go#L418-L440